docs/developer-guide/ksqldb-reference/select-pull-query.md

```diff
@@ -14,98 +14,125 @@ Synopsis
+  FROM table
+  [ WHERE where_condition ]
+  [ AND window_bounds ];
 ```
 
 Description
 -----------
 
-Pulls the current value from the materialized table and terminates. The result
-of this statement isn't persisted in a {{ site.ak }} topic and is printed out
-only in the console.
-
-Pull queries enable you to fetch the current state of a materialized view.
-Because materialized views are incrementally updated as new events arrive,
-pull queries run with predictably low latency. They're a great match for
-request/response flows. For asynchronous application flows, see
+Pulls the current value from the materialized view and terminates. The result
+of this statement is not persisted in a {{ site.ak }} topic and is printed out
+only in the console. Because materialized views are incrementally updated as new 
+events arrive, pull queries run with predictably low latency. They are a great match 
+for request/response flows. For asynchronous application flows, see 
 [Push Queries](select-push-query.md).
 
-Execute a pull query by sending an HTTP request to the ksqlDB REST API, and
+You can execute a pull query by sending an HTTP request to the ksqlDB REST API, and
 the API responds with a single response.  
 
-The WHERE clause must contain a value for each primary-key column to retrieve and may
-optionally include bounds on `WINDOWSTART` and `WINDOWEND` if the materialized table is windowed.
-For more information, see 
-[Time and Windows in ksqlDB](../../concepts/time-and-windows-in-ksqldb-queries.md).
+You can issue a pull query against any table that was created by using 
+a [CREATE TABLE AS SELECT](../../ksqldb-reference/create-table-as-select) 
+statement. Currently, we do not support pull queries against tables created 
+by using a [CREATE TABLE](../../ksqldb-reference/create-table) statement.
+
+WHERE Clause Guidelines
+-----------------------
+
+By default, only key lookups are enabled. They have the following requirements:
+-   Key column(s) must use an equality comparison to a literal (e.g. KEY = 'abc').
+-   On windowed tables, WINDOWSTART and WINDOWEND can be optionally compared to literals. 
+    For more information on windowed tables, see [Time and Windows in ksqlDB](../../concepts/time-and-windows-in-ksqldb-queries.md).
+
+You can enable table scans to loosen the restrictions on the `WHERE` clause or eliminate 
+the `WHERE` clause altogether. Table scans can be turned on for pull queries running in the 
+current CLI session with the command `SET 'ksql.query.pull.table.scan.enabled'='true';`. 
+They can also be enabled by default by setting a server configuration property with 
+`ksql.query.pull.table.scan.enabled=true`.Once table scans are enabled, the following 
+additional expressions are allowed:
+-   Key column(s) using range comparisons to literals.
+-   Non key columns to be used alone, without key references.
+-   Columns to be compared to other columns.
+-   References to subsets of columns from a multi-column key.

```

Comment:
Forgot to add this:
- Complex expressions without direct column references (e.g. `instr(NAME_COL, 'hello') > 0`)

Reply:
added